pennylane/workflow/qnode.py

```diff
@@ -28,6 +28,7 @@
 from pennylane.logging import debug_logger
 from pennylane.measurements import CountsMP, MidMeasureMP, Shots
 from pennylane.tape import QuantumScript, QuantumTape
+from pennylane.transforms.core.transform_program import _prune_dynamic_transform

```

Comment:
I'd prefer not to have a coupling between the transform program and a high level workflow class via a private method.  I know `prune_dynamic_transform` is kinda a patch, but I do think it needs to be a public patch.

Reply:
How about I move this function into `_qnode.py`? The transform program shouldn't need to know what inner transform and outer transform are anyways?

Reply:
I actually like that idea 👍 This does seem to be more about "how do I setup and validate a workflow" than maintaining and executing a generic queue of transforms.